sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```
@@ -88,4 +106,38 @@ public long getRecordTimestamp(ConsumerRecord<byte[], byte[]> rawRecord) {
     }
     return timestamp;
   }
+
+  @SuppressWarnings("unchecked")
+  public long offsetForTime(Consumer<?, ?> consumer, TopicPartition topicPartition, Instant time) {
+
+    if (!hasOffsetsForTimes) {
```

This check can be move to `Read.validate()` so that pipeline fails early.